spec/models/room_spec.rb

```diff
@@ -43,4 +43,16 @@
       expect(room.to_param).to eq room.public_id
     end
   end
+
+  describe '#self.random_id' do
```

Comment:
From here you can instead write a spec in `spec/requests/controller/` to ensure a redirect happens.